scripts/build.mjs

```diff
@@ -49,10 +54,44 @@ async function buildNodePackages() {
   }
 
   for (const {packageDir, pkg} of buildConfigs) {
+    const entryPointFile = path.resolve(packageDir, pkg.main);
+
     assert.ok(
-      fs.existsSync(path.resolve(packageDir, pkg.main)),
+      fs.existsSync(entryPointFile),
       `Main file "${pkg.main}" in "${pkg.name}" should exist`,
     );
+
+    // type only packages
+    if (typeOnlyPackages.has(pkg.name)) {
+      continue;
+    }
+
+    const entryPoint = require(entryPointFile);
+    const exportStatements = Object.keys(entryPoint)
+      .filter(name => name !== '__esModule' && name !== 'default')
+      .map(name => `export const ${name} = entryPoint.${name};`);
+
+    if (entryPoint.default) {
+      exportStatements.push('export default entryPoint.default;');
+    }
+
+    if (exportStatements.length === 0) {
+      throw new Error(`No exports found in package ${pkg.name}`);
+    }
+
+    const mjsEntryFile = entryPointFile.replace(/\.js$/, '.mjs');
+
+    const esSource = dedent`
+      import {createRequire} from 'module';
+
+      const require = createRequire(import.meta.url);
+
+      const entryPoint = require('./index.js');
```

Comment:
not sure how bundlers deal with `createRequire` calls... we might have to adjust if people say it breaks them 😀 

Maybe top level `await import('./index.js')` is better